apps/web/src/scenes/UserGalleryPage/UserSharedCommunities.tsx

```diff
@@ -0,0 +1,127 @@
+import { useCallback, useMemo } from 'react';
```

Comment:
a lot of the logic is duplicated from `UserSharedCommunities`

Ideally we can share some pieces, but I found that because the data type being displayed is different (user vs community) and there are enough small differences between the two, making the logic shareable made it more difficult to follow and didn't provide a clear benefit. 

Since we only have 2 versions, I think this is acceptable. maybe if we add a 3rd similar category we can reconsider consolidating the shared logic.

Reply:
that being said if anyone sees any obvious things that could be reused pls lmk!

Reply:
totally chill, way easier to understand this way when they're just explicitly split up